### ISSUE TITLE
Sampler.Uniform sampler crashes with TypeError: randint()

### ISSUE THREAD

**Issue by hawkeyedesi** [fepegar/torchio#374]
**🐛Bug**
Great tookit. Thank you for that. When running the tio.Sampler.UniformSampler(), I get the following error: 

`TypeError: randint(): argument 'high' (position 1) must be int, not numpy.int64` on this line here: 
https://github.com/fepegar/torchio/blob/e0b37b77052f7342e4cab5a8cb045f91664dc58e/torchio/data/sampler/uniform.py#L33

**To reproduce**
The error occurs when I am trying to reproduce the Colab example: 
Please see this gist for a subset of the code which causes the error: https://gist.github.com/hawkeyedesi/91d7f1d3b4be7ad85124fd95423d53bb

```python-traceback

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-9-d3f26251d853> in <module>
----> 1 one_batch = next(iter(training_loader))
      2 k = int(patch_size // 4)
      3 batch_mri = one_batch['mri'][DATA][..., k]
      4 batch_label = one_batch['brain'][DATA][..., k]
      5 slices = torch.cat((batch_mri, batch_label))

~/anaconda3/envs/venv_pytorch/lib/python3.7/site-packages/torch/utils/data/dataloader.py in __next__(self)
    344     def __next__(self):
    345         index = self._next_index()  # may raise StopIteration
--> 346         data = self._dataset_fetcher.fetch(index)  # may raise StopIteration
    347         if self._pin_memory:
    348             data = _utils.pin_memory.pin_memory(data)

~/anaconda3/envs/venv_pytorch/lib/python3.7/site-packages/torch/utils/data/_utils/fetch.py in fetch(self, possibly_batched_index)
     42     def fetch(self, possibly_batched_index):
     43         if self.auto_collation:
---> 44             data = [self.dataset[idx] for idx in possibly_batched_index]
     45         else:
     46             data = self.dataset[possibly_batched_index]

~/anaconda3/envs/venv_pytorch/lib/python3.7/site-packages/torch/utils/data/_utils/fetch.py in <listcomp>(.0)
     42     def fetch(self, possibly_batched_index):
     43         if self.auto_collation:
---> 44             data = [self.dataset[idx] for idx in possibly_batched_index]
     45         else:
     46             data = self.dataset[possibly_batched_index]

~/anaconda3/envs/venv_pytorch/lib/python3.7/site-packages/torchio/data/queue.py in __getitem__(self, _)
    107         if not self.patches_list:
    108             self._print('Patches list is empty.')
--> 109             self.fill()
    110         sample_patch = self.patches_list.pop()
    111         self.num_sampled_patches += 1

~/anaconda3/envs/venv_pytorch/lib/python3.7/site-packages/torchio/data/queue.py in fill(self)
    164             subject = self.get_next_subject()
    165             iterable = self.sampler(subject)
--> 166             patches = list(islice(iterable, self.samples_per_volume))
    167             self.patches_list.extend(patches)
    168         if self.shuffle_patches:

~/anaconda3/envs/venv_pytorch/lib/python3.7/site-packages/torchio/data/sampler/uniform.py in __call__(self, subject)
     31         valid_range = subject.spatial_shape - self.patch_size
     32         while True:
---> 33             index_ini = [torch.randint(x + 1, (1,)).item() for x in valid_range]
     34             index_ini_array = np.asarray(index_ini)
     35             yield self.extract_patch(subject, index_ini_array)

~/anaconda3/envs/venv_pytorch/lib/python3.7/site-packages/torchio/data/sampler/uniform.py in <listcomp>(.0)
     31         valid_range = subject.spatial_shape - self.patch_size
     32         while True:
---> 33             index_ini = [torch.randint(x + 1, (1,)).item() for x in valid_range]
     34             index_ini_array = np.asarray(index_ini)
     35             yield self.extract_patch(subject, index_ini_array)

TypeError: randint(): argument 'high' (position 1) must be int, not numpy.int64
```

**Expected behavior**

To run the patchwise images through the tio.Queue and display the patchwise examples. 

**Actual behavior**

Code breaks when calling the UniformSampler. 

**TorchIO version**
TorchIO version: 0.18.1

**Comment by fepegar** [comment]
Hi, @hawkeyedesi. Thanks for reporting and for the gist. I haven't been able to replicate on the original tutorial nor locally, and the gist is unfortunately not self-contained. Can you please share the link of the tutorial and tell me what exact steps I need to follow to get that error?

**Comment by hawkeyedesi** [comment]
Hi @fepegar thanks for the feedback. 

>I haven't been able to replicate on the original tutorial nor locally, and the gist is unfortunately not self-contained.

I have modified the gist and made it self contained (using random dataset).
Gist Link: https://gist.github.com/hawkeyedesi/5a4995fb0661d5aa3561d7a8e2e72ade

>Can you please share the link of the tutorial and tell me what exact steps I need to follow to get that error?

I was following this tutorial: https://colab.research.google.com/github/fepegar/torchio-notebooks/blob/main/notebooks/TorchIO_tutorial.ipynb

On Google Colab it ran correctly without errors. 

Steps to recreate the error: 
- I download the jupyter notebook and ran it on my local environment 
- The gist file has a minimum example of the TorchIO_tutorial.ipynb to recreate the error. 
- Attached is the conda yml file that I used when debugging this. 
- I am running this on a Linux Box. 

Thanks. 
I hope this helps. 

[torchio_debug.yml.zip](https://github.com/fepegar/torchio/files/5662201/torchio_debug.yml.zip)
